### PR TITLE
Redirect users to Cloud after checking out

### PR DESCRIPTION
**Pull request by ChaosExAnima**
#### Changes proposed in this Pull Request

* Redirects users to the appropriate place after purchasing a Cloud product.
* Ensures external redirects work correctly for checkouts.

#### Testing instructions

1. Check out with Jetpack Backup or Scan.
2. See where you are directed after checkout.

### CODE REVIEW

**Comment by ChaosExAnima** [comment]
One thing I've noticed: this redirect happens for renewals as well as original purchases. This may not be desired behavior?

File: client/my-sites/checkout/checkout/index.jsx [hunk 1]
```diff
@@ -541,14 +547,25 @@ export class Checkout extends React.Component {
...
+			if ( isJetpackBackupSlug( product ) ) {
+				source = 'calypso-backups';
+			} else if ( isJetpackScanSlug( product ) ) {
+				source = 'calypso-scanner';
+			}
+			if ( source ) {
+				return addQueryArgs(
+					{
+						source,
+						site: selectedSiteSlug,
```
Comment (rcoll) [inline client/my-sites/checkout/checkout/index.jsx:563]: Will the Jetpack redirects work just the same with a site slug versus a blog id?

File: client/lib/products-values/index.js [hunk 2]
```diff
@@ -234,6 +234,12 @@ export function isJetpackScan( product ) {
 	return isJetpackScanSlug( product.product_slug );
 }
 
+export function isJetpackCloudProductSlug( productSlug ) {
```
Comment (rcoll) [inline client/lib/products-values/index.js:237]: Not a blocker, but it seems like the function should be named a bit differently, something like `isJetpackCloudEligibleSlug`. Again, not a blocker.

**Comment by rcoll** [comment]
`One thing I've noticed: this redirect happens for renewals as well as original purchases. This may not be desired behavior?`

I think that's probably fine for now.

**Comment by rcoll** [comment]
Tests GOOD! :ship:

**Comment by sirbrillig** [comment]
Coming in a bit late but 👍 😁

File: client/my-sites/checkout/checkout/index.jsx (hunk 1 above)
Comment (ChaosExAnima) [reply client/my-sites/checkout/checkout/index.jsx:563]: Yep! They did in my testing, at least.

**Pull request merged**